config/identity/handler/default.json

```diff
@@ -21,7 +21,8 @@
       "providerFactory": { "@id": "urn:solid-server:default:IdentityProviderFactory" },
       "interactionPolicy": { "@id": "urn:solid-server:auth:password:AccountInteractionPolicy" },
       "interactionHttpHandler": { "@id": "urn:solid-server:auth:password:InteractionHttpHandler" },
-      "errorResponseWriter": { "@type": "ErrorResponseWriter" }
+      "errorHandler": { "@id": "urn:solid-server:default:ErrorHandler" },
+      "responseWriter": { "@id": "urn:solid-server:default:ResponseWriter" }
```

Comment:
I agree actually, that's what I meant with "extract the preferences at an earlier point" above. It might make sense to extract the preferences in the `BaseHttpServerFactory` and catch all errors there, instead of doing it in the HttpHandlers. This way we have a single place where error responses are handled. The only reason this was also added now to the IDP is because I previously changed it to use an error responsewriter instead of just returning the error. The `StaticAssetHandler` doesn't have this yet for example so just throws errors to the server factory.

The one disadvantage would be that then there would be no way for different HttpHandler to have different metadata writers for errors that are thrown. Currently this is not needed yet though. 

We could also keep the current solution but then we probably want to, at least for the IDP, find a way to extract preferences there so you don't get a text error while going through the html UI.

Reply:
Additonal confusion: the current `BaseHttpServerFactory` also has a try/catch for errors which writes out the error stack (which is how the StaticAssetHandler errors get returned). So there is some duplication where errors are handled.

Reply:
Let's keep it as-is. It keeps the different subsystems independent.

However, let's move (in another PR, or this one)

```
      const preferences: RepresentationPreferences = { type: { 'text/plain': 1 }};
      const result = await this.errorHandler.handleSafe({ error, preferences });
      await this.responseWriter.handleSafe({ response: input.response, result });
```

to a single class, which is then the `errorWriter` that we pass (instead of `errorHandler` and `responseWriter`).
That `errorWriter` can then also have a preferences extractor.

---

> the current `BaseHttpServerFactory` also has a try/catch for errors which writes out the error stack

That's the (server-specific) emergency solution for when even error handling fails; let's keep that in place.

---

Just a small note for future reference:

> It might make sense to extract the preferences in the `BaseHttpServerFactory`

This factory should really only make the main server.
Extracting preferences can be done as just an HTTP handler that is reusable with any server.